### ISSUE TITLE
Remove duplicit docstrings + make documentation easier to navigate

### ISSUE THREAD

**Issue by mfocko** [packit/ogr#533]
pdoc(3) supports inheriting of docstrings, there is no need to have multiple identical docstrings and also try to rethink exporting of the classes/modules through `__all__`, since it makes the documentation harder to navigate

- [ ] remove duplicit docstrings
   - [ ] add specific docstrings where the interface deviates
- [ ] try to think about better structure of exports for `__all__` (shouldn't be a problem in specific services, just clean it up at the top-level)

Follow-up of #531

**Comment by lachmanfrantisek** [comment]
@mfocko , you are working on this, aren't you?

**Comment by mfocko** [comment]
> @mfocko , you are working on this, aren't you?

yeah, slowly... didn't even realize how much docs do we have in ogr :D

**Comment by lachmanfrantisek** [comment]
Reopening, since @mfocko already has spent some time on it.

**Comment by mfocko** [comment]
@lachmanfrantisek I should open PR and finish it finally :no_mouth:
